### ISSUE TITLE
rename table to anothor database causing insert to fail

### ISSUE THREAD

**Issue by Darren** [pingcap/tidb#4858]
1. What did you do?

There's some history about rename schema name in MySQL as described in this blog:
https://www.percona.com/blog/2013/12/24/renaming-database-schema-mysql/

It seems that TiDB doen't support rename database name in a simple way either.

Because we tend to create all databases in one TiDB cluster,  we must choose schema carefully, but if name clash actually happens, we have to rename the database name.

So I tried to rename database as described in above blog using syntax as: `rename table db1.t1  to  db2.t1`

this is a simple test case to reproduce:
```sql

create database db1;
create database db2;
create table db1.t1 (id int);
rename table db1.t1 to  db2.t1;
insert into db2.t1 values (1);
use db2;
insert into t1 values (1);
```

2. What did you expect to see?

the moved table should work properly

3. What did you see instead?

I can not insert data into the new table any longer (while select is OK)

```sql
TiDB-Server [test]> create database db1;
Query OK, 0 rows affected (1.25 sec)

TiDB-Server [test]> create database db2;
Query OK, 0 rows affected (1.21 sec)

TiDB-Server [test]> create table db1.t1 (id int);
Query OK, 0 rows affected (1.32 sec)

TiDB-Server [test]> rename table db1.t1 to  db2.t1;
Query OK, 0 rows affected (1.17 sec)

TiDB-Server [test]> insert into db2.t1 values (1);
ERROR 1146 (42S02): table doesn't exist

TiDB-Server [test]> use db2
Database changed
TiDB-Server [db2]> show tables;
+---------------+
| Tables_in_db2 |
+---------------+
| t1            |
+---------------+
1 row in set (0.00 sec)

TiDB-Server [db2]> insert into t1 values (1);
ERROR 1146 (42S02): table doesn't exist
```

4. What version of TiDB are you using (`tidb-server -V`)?

Release Version: 1.0.0
Git Commit Hash: 21c21973c092e3e192c0210933448d7b394e74cb
Git Branch: master
UTC Build Time: 2017-10-17 01:35:28

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
@Darren Thanks for your report! We'll fix it later.

**Comment by Darren** [comment]
@shenli @zimulala 

Will TiDB support `rename database db1 to db2` in the future?

**Comment by zimulala** [comment]
Yes, we will. @Darren

**Comment by Darren** [comment]
@shenli  @zimulala 

Since this bug exists in the released 1.0.0 version, I want to know how to fix the already renamed the table before #4862?  

Is there any way to fix the table already in question?

I tried rename the table and rename back:

```
rename table bad_1 to bad_1_temp;
rename table bad_1_temp to bad_1;
```

But when trying insert data to the table, it shows `ERROR 1105 (HY000): Error: key not exist`

```log
2017/10/26 15:08:46.472 conn.go:427: [warning] [1] dispatch error:
id:1, addr:192.168.20.8:52884 status:2, collation:utf8_general_ci, user:root
[kv:2]Error: key not exist
/go/src/github.com/pingcap/tidb/meta/autoid/autoid.go:129: 
/go/src/github.com/pingcap/tidb/kv/txn.go:54: 
/go/src/github.com/pingcap/tidb/meta/autoid/autoid.go:135: 
/go/src/github.com/pingcap/tidb/table/tables/tables.go:330: 
/go/src/github.com/pingcap/tidb/executor/write.go:766: 
/go/src/github.com/pingcap/tidb/executor/adapter.go:207: 
/go/src/github.com/pingcap/tidb/tidb.go:171: 
/go/src/github.com/pingcap/tidb/session.go:669: 
/go/src/github.com/pingcap/tidb/session.go:746: 
/go/src/github.com/pingcap/tidb/server/conn.go:733: 
2017/10/26 15:08:46.472 terror.go:281: [warning] Unknown error class: kv code: 2
```

Release Version: 1.0.0
Git Commit Hash: e80e4c6137abfd149e0c28df46202f8fabf824f4
Git Branch: master
UTC Build Time: 2017-10-26 06:56:42
